__tests__/textalyze.test.js

```diff
@@ -82,3 +82,45 @@ describe('sanitize', () => {
     expect(sanitize(input)).toEqual(expectedOutput);
   });
 });
+
+describe('updateToFrequencyStatistics', () => {
+  test('returns the percentage', () => {
+    const inputMap = new Map([['a', 2], ['A', 2]]);
+    const inputTotal = 4;
+    const expectedOutput = new Map([['a', 0.5], ['A', 0.5]]);
+
+    expect(getFrequencyStatistics(inputMap, inputTotal)).toEqual(expectedOutput);
+  });
+
+  test('handles non map input', () => {
+    const inputMap = 123456;
+    const inputTotal = 4;
+    const expectedOutput = 123456;
+
+    expect(getFrequencyStatistics(inputMap, inputTotal)).toEqual(expectedOutput);
```

Comment:
I'm glad you're testing the specific conditionals you added to your function, but this behavior is very surprising to me.

In terms of function signatures, I'd expect it to be one of the following:

```
type CountMap = Map<string, number>

getFrequencyStatistics :: (CountMap, number) -> CountMap | null
getFrequencyStatistics :: (CountMap, number) -> CountMap | throws Error
getFrequencyStatistics :: (CountMap, number) -> CountMap
```

Here `CountMap | throws Error` means "returns a `CountMap` or throws an exception".

If we're guaranteeing that `CountMap` always returns and never returns `null` then the only thing that makes sense to me is the empty Map (`new Map()`).

If your signature is

```
getFrequencyStatistics :: (CountMap, number) -> CountMap | number
```

then every function that accepts the output of `getFrequencyStatistics` needs to accept a `CountMap | number` as input.  But you don't do that:

```javascript
const statistics = getFrequencyStatistics(...);

statistics.forEach(...);
```

If `statistics` is of type `number` then `statistics.forEach` is going to throw an error along the lines of:

```
TypeError: statistics.forEach is not a function
```

This error message is not helpful.  Instead, consider something like the following:

```
function getType(obj) {
  if (obj === null) { return 'null'; }
  if (obj === undefined) { return 'undefined'; }

  return obj.constructor.name;
}

function getFrequencyStatistics(counts, total) {
  if (!(counts instanceof Map)) {
    throw new TypeError(`Argument must be a Map, received a '${getType(counts)}'.`);
  }
}
```

You're still throwing an error — something that would happen anyways — but the error is more useful in at least two ways:

1. The error message clearly indicates what is going wrong and the stack trace will point to the exact line where the problem is
1. Calling code can catch `TypeError` to print out something more useful to the end-user.  By the time we get to `statistics.forEach` and catch an exception, it's impossible for us to tell what the problem is and do anything useful for the user.